File: LC/[1143] Longest Common Subsequence.cpp
class Solution {
    public:
        int longestCommonSubsequence(string text1, string text2) {
            int N1 = text1.size();
            int N2 = text2.size();
    
            string ss = "#" + text1;
            string tt = "#" + text2;
    
            auto dp = vector<vector<int>>(N1+1, vector<int>(N2+1, 0));
    
            for(int ii=1; ii<=N1; ++ii){
                for(int jj=1; jj<=N2; ++jj){
                    if(ss[ii]==tt[jj]){
                        dp[ii][jj] = max(dp[ii][jj], dp[ii-1][jj-1]+1);
                    }else{
                        dp[ii][jj] = max(dp[ii][jj], max(dp[ii-1][jj], dp[ii][jj-1]));
                    }
                }
            }
    
    
            return dp[N1][N2];
    
        }
    };

File: LC/[1143] Longest Common Subsequence.java
class Solution {
    public int longestCommonSubsequence(String text1, String text2) {
        int N1 = text1.length();
        int N2 = text2.length();

        String ss="#"+text1;
        String tt="#"+text2;

        var dp = IntStream
                .range(0, N1+1)
                .mapToObj(ii -> IntStream.range(0, N2+1).map(jj->0).toArray())
                .toArray(int[][]::new);

        for(int ii=1; ii<=N1; ++ii){
            for(int jj=1; jj<=N2; ++jj){
                if(ss.charAt(ii)==tt.charAt(jj)){
                    dp[ii][jj] = dp[ii-1][jj-1] + 1;
                }else{
                    dp[ii][jj] = Math.max(dp[ii-1][jj], dp[ii][jj-1]);
                }
            }
        }

        return dp[N1][N2];
    }
}